### ISSUE TITLE
Referencing documents are not working if .md extension is used

### ISSUE THREAD

**Issue by amimas** [facebook/Docusaurus#1261]
## 🐛 Bug Report

When referencing other docs using `.md` extension, Docusaurus is failing to generate a proper link.

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## To Reproduce

(Write your steps here:)

1. Follow the steps mentioned here: https://docusaurus.io/docs/en/navigation#referencing-site-documents
1. Create 2 documents (`doc1.md`, `doc2.md`)
1. Create a link in `doc1.md` (`[link to document 2](doc2.md)`)

## Expected behavior

The above steps should generates a proper link to `doc2.html` in both dev server and also when the site is built for publishing

## Actual Behavior

The browser shows an error when clicking on the link in `doc1.html`

```
Cannot GET /test-site/docs/doc2.md
```

When building the site, docusaurus also throws following warnings:

```
[WARN] unresolved links in file 'doc1.md' > [ 'doc2.md' ]
```

## Reproducible Demo

I will try to create a reproducible demo soon. But, the steps mentioned above are very simple.

**Comment by amimas** [comment]
This issue was also brought up before in #1137  but not sure why it was closed.

**Comment by amimas** [comment]
Ooops. Just found my mistake. I had a typo in the doc file name that was being linked to.
Closing it.

**Comment by endiliey** [comment]
#1137 was fixed by #1138 and hence got closed. I believe this might be a regression.
